devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -605,10 +626,42 @@ bool YarpRobotLoggerDevice::setupRobotCameraBridge(
     return true;
 }
 
+bool YarpRobotLoggerDevice::initMetadata(const std::string& nameKey, const std::vector<std::string>& metadataNames)
+{
+    bool ok = m_bufferManager.addChannel({nameKey, {metadataNames.size(), 1}, metadataNames});
+    std::string rtNameKey = robotRtRootName + treeDelim + nameKey;
+    if (m_sendDataRT)
+    {
+        ok &= m_vectorCollectionRTDataServer.populateMetadata(rtNameKey, metadataNames);
+    }
+    return ok;
+}
+
+void YarpRobotLoggerDevice::logData(const std::string& name,
+                                                    const Eigen::VectorXd& data,
+                                                    const double time)
```

Comment:
Fixed in [0888c96](https://github.com/ami-iit/bipedal-locomotion-framework/commit/0888c9631b6c934ab69fce248eae0c9587910fc7). The change of just adding Eigen::Ref<> causes issues when saving the data via buffer manager. This is because the push_back method of the buffer manager uses a template for the data and is storing the raw data of the Eigen::Ref and not the Eigen::VectorXd. The solution was discussed and implemented with the help of @S-Dafarra in which we used a lambda function to still pass the Eigen::Vector by a reference.